### PR TITLE
power up cloud function

### PR DESCRIPTION
**Pull request by r-temp**
add function to give user money at noon
take double vote power up into account when counting votes
deduce the price of the power up from the section funds after each usage

### CODE REVIEW

**Review by Ant-One — APPROVED** [review]
Really nice work! Now we can try this in real situations :)

File: functions/index.js [hunk 1]
```diff
@@ -72,72 +73,171 @@ exports.resetVotesScheduledFunction = functions.region('europe-west1').pubsub.sc
...
+    const db = getDatabase()
+    const refUsers = db.ref('Users')
+
+    //take snapshot of Users
+    refUsers.once('value').then( (usersSnapshot) => {
 
-                sectionsScores.set(current_owner, sectionsScores.get(current_owner) + 1)
+        //count number of user for each section
+        const sectionsUserCount = new Map()
+        for(var i = 0, size = sections.length; i < size ; i++){
```
Comment (Ant-One) [inline functions/index.js:191]: Could collapse this loop with the one below

**Review by Ant-One — CHANGES_REQUESTED** [review]
Just these two loops that are similar, and maybe you could merge them?

**Comment by Jenjamin3000** [comment]
Be aware to merge with the main on your branch before merging the PR! There is for example TOS that has been removed and SectionColors that has been tested.

**Comment by Ant-One** [comment]
@Jenjamin3000 it should not matter, because @r-temp has not modified theses in his PR, should no code could be lost :)

**Review by Ant-One — APPROVED** [review]

File: functions/index.js (hunk 1 above)
Comment (r-temp) [reply functions/index.js:191]: I think bc the second one add 1 to an existing value, we need to initalize the values with either a loop before or a condition in the second one (both are 3 loc)

**Pull request merged**